examples/advanced/cugraph/graphsage.py

```diff
@@ -0,0 +1,200 @@
+import argparse
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torchmetrics.functional as MF
+import tqdm
+from dgl.data import AsNodePredDataset
+from dgl.dataloading import (
+    DataLoader,
+    MultiLayerFullNeighborSampler,
+    NeighborSampler,
+)
+from dgl.nn import CuGraphSAGEConv
+from ogb.nodeproppred import DglNodePropPredDataset
+
+
+class SAGE(nn.Module):
+    def __init__(self, in_size, hid_size, out_size):
+        super().__init__()
+        self.layers = nn.ModuleList()
+        # three-layer GraphSAGE-mean
+        self.layers.append(CuGraphSAGEConv(in_size, hid_size, "mean"))
+        self.layers.append(CuGraphSAGEConv(hid_size, hid_size, "mean"))
+        self.layers.append(CuGraphSAGEConv(hid_size, out_size, "mean"))
+        self.dropout = nn.Dropout(0.5)
+        self.hid_size = hid_size
+        self.out_size = out_size
+
+    def forward(self, blocks, x):
+        h = x
+        for l, (layer, block) in enumerate(zip(self.layers, blocks)):
+            h = layer(block, h, max_in_degree=10)
```

Comment:
This seems a bit hacky.

Reply:
Can you explain what needs to be done here? Are you suggesting to unpack to loop like this?
```python
h = F.relu(self.conv1(g[0], x))
h = F.relu(self.conv2(g[1], h))
...
```

Reply:
I meant the specification of `max_in_degree`.

Reply:
I see and I do agree that it is not an ideal interface. We did not make `max_in_degree` an attribute of `CuGraphSAGEConv` since it is a property of the graph (i.e., `block`), rather than the model. I have removed it from the example as this flag is optional. 
In the meantime, we are improving our aggregation primitives to be more flexible to eventually ditch this option.